ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/SyncStepStep.java

```diff
@@ -39,27 +44,44 @@ public SyncStepStep(final BackwardSyncContext context, final BackwardChain backw
   }
 
   public CompletableFuture<Block> executeAsync(final Hash hash) {
-    return CompletableFuture.supplyAsync(() -> hash)
-        .thenCompose(this::requestBlock)
-        .thenApply(this::saveBlock);
-  }
-
-  private CompletableFuture<Block> requestBlock(final Hash targetHash) {
-    LOG.atDebug().setMessage("Fetching block by hash {} from peers").addArgument(targetHash).log();
-    final RetryingGetBlockFromPeersTask getBlockTask =
-        RetryingGetBlockFromPeersTask.create(
-            context.getProtocolSchedule(),
-            context.getEthContext(),
-            context.getSynchronizerConfiguration(),
-            context.getMetricsSystem(),
-            context.getEthContext().getEthPeers().peerCount(),
-            Optional.of(targetHash),
-            UNUSED);
     return context
         .getEthContext()
         .getScheduler()
-        .scheduleSyncWorkerTask(getBlockTask::run)
-        .thenApply(AbstractPeerTask.PeerTaskResult::getResult);
+        .scheduleServiceTask(
+            () -> {
+              Block block = requestBlock(hash);
+              saveBlock(block);
+              return CompletableFuture.completedFuture(block);
+            });
+  }
+
+  private Block requestBlock(final Hash targetHash) {
+    LOG.debug("Fetching block by hash {} from peers", targetHash);
+    GetHeadersFromPeerTask headersFromPeerTask =
+        new GetHeadersFromPeerTask(
+            targetHash,
+            UNUSED,
```

Comment:
Would prefer a different constructor rather than using a sentinel value for the block number